src/sidebar/components/annotation-share-info.js

```diff
@@ -6,12 +6,18 @@ import { isPrivate } from '../util/permissions';
 
 import SvgIcon from '../../shared/components/svg-icon';
 
+/**
+ * @typedef {import('../../types/api').Group} Group
+ */
+
 /**
  * Render information about what group an annotation is in and
  * whether it is private to the current user (only me)
  */
 function AnnotationShareInfo({ annotation }) {
-  const group = useStore(store => store.getGroup(annotation.group));
+  const group = /** @type {Group} */ (useStore(store =>
```

Comment:
it really barked at me if I took this cast out. It does not seem to be able to parse the jsx below where it checks `hasGroup` first, then `group.blah`. I'm not sure I understand why or perhaps I'm looking at it wrong. A simple solution here is to cast this as it can't fail because we're checking `hasGroup` before any other member referencing of that potentially `undefined` object

Reply:
> it really barked at me if I took this cast out. It does not seem to be able to parse the jsx below where it checks hasGroup first, then group.blah

It seems null-ness inference is not transitive:

```js
if (group) {
 // `group` is not null here
}
```

But if there is some indirection:

```js
const hasGroup = !!group;
if (hasGroup) {
  // TS thinks `group` may be null here
}
```

In this case a way around this is to remove the `hasGroup` variable and test the group more directly:

```diff
diff --git a/src/sidebar/components/annotation-share-info.js b/src/sidebar/components/annotation-share-info.js
index dbbfb6225..11041e7f7 100644
--- a/src/sidebar/components/annotation-share-info.js
+++ b/src/sidebar/components/annotation-share-info.js
@@ -15,22 +15,17 @@ import SvgIcon from '../../shared/components/svg-icon';
  * whether it is private to the current user (only me)
  */
 function AnnotationShareInfo({ annotation }) {
-  const group = /** @type {Group} */ (useStore(store =>
-    store.getGroup(annotation.group)
-  ));
-
-  // We may not have access to the group object beyond its ID
-  const hasGroup = !!group;
+  const group = useStore(store => store.getGroup(annotation.group));
 
   // Only show the name of the group and link to it if there is a
   // URL (link) returned by the API for this group. Some groups do not have links
-  const linkToGroup = hasGroup && group.links && group.links.html;
+  const linkToGroup = group?.links.html;
 
   const annotationIsPrivate = isPrivate(annotation.permissions);
 
   return (
     <div className="annotation-share-info u-layout-row--align-baseline">
-      {linkToGroup && (
+      {group && linkToGroup && (
         <a
           className="u-layout-row--align-baseline u-color-text--muted"
           href={group.links.html}
diff --git a/src/sidebar/components/test/annotation-share-info-test.js b/src/sidebar/components/test/annotation-share-info-test.js
index d1bd20fd1..4d599d26c 100644
--- a/src/sidebar/components/test/annotation-share-info-test.js
+++ b/src/sidebar/components/test/annotation-share-info-test.js
@@ -108,7 +108,7 @@ describe('AnnotationShareInfo', () => {
       });
 
       it('should show "only me" text for annotation in third-party group', () => {
-        fakeGetGroup.returns({ name: 'Some Name' });
+        fakeGetGroup.returns({ name: 'Some Name', links: {} });
         const wrapper = createAnnotationShareInfo();
 
         const privacyText = wrapper.find(
```

The above diff does feature a redundant test of whether `group` is null in the JSX, but this is very cheap compared to everything else that is going on, and it avoids a problem if `linkToGroup` is changed in future in such a way that it may be set even if `group` is null.

Looking at the `h` backend, the `group.links` object should always exist, but it may be empty. Therefore I adjusted a test fixture accordingly.